chirp/drivers/uvk5.py

```diff
@@ -0,0 +1,1972 @@
+
+    # Return a raw representation of the memory object, which
+    # is very helpful for development
+    def get_raw_memory(self, number):
+        return repr(self._memobj.channel[number-1])
+
+    def validate_memory(self, mem):
+        msgs = super().validate_memory(mem)
+        return msgs
+
+    def _set_tone(self, mem, _mem):
+        ((txmode, txtone, txpol),
+         (rxmode, rxtone, rxpol)) = chirp_common.split_tone_encode(mem)
+
+        if txmode == "Tone":
+            txtoval = CTCSS_TONES.index(txtone)
+            txmoval = 0b01
+        elif txmode == "DTCS":
+            txmoval = txpol == "R" and 0b11 or 0b10
+            txtoval = DTCS_CODES.index(txtone)
+        else:
+            txmoval = 0
+            txtoval = 0
+
+        if rxmode == "Tone":
+            rxtoval = CTCSS_TONES.index(rxtone)
+            rxmoval = 0b01
+        elif rxmode == "DTCS":
+            rxmoval = rxpol == "R" and 0b11 or 0b10
+            rxtoval = DTCS_CODES.index(rxtone)
+        else:
+            rxmoval = 0
+            rxtoval = 0
+
+        _mem.rxcodeflag = rxmoval
+        _mem.txcodeflag = txmoval
+        _mem.unknown1 = 0
+        _mem.unknown2 = 0
+        _mem.rxcode = rxtoval
+        _mem.txcode = txtoval
+
+    def _get_tone(self, mem, _mem):
+        rxtype = _mem.rxcodeflag
+        txtype = _mem.txcodeflag
+        rx_tmode = TMODES[rxtype]
+        tx_tmode = TMODES[txtype]
+
+        rx_tone = tx_tone = None
+
+        if tx_tmode == "Tone":
+            if _mem.txcode < len(CTCSS_TONES):
+                tx_tone = CTCSS_TONES[_mem.txcode]
+            else:
+                tx_tone = 0
+                tx_tmode = ""
+        elif tx_tmode == "DTCS":
+            if _mem.txcode < len(DTCS_CODES):
+                tx_tone = DTCS_CODES[_mem.txcode]
+            else:
+                tx_tone = 0
+                tx_tmode = ""
+
+        if rx_tmode == "Tone":
+            if _mem.rxcode < len(CTCSS_TONES):
+                rx_tone = CTCSS_TONES[_mem.rxcode]
+            else:
+                rx_tone = 0
+                rx_tmode = ""
+        elif rx_tmode == "DTCS":
+            if _mem.rxcode < len(DTCS_CODES):
+                rx_tone = DTCS_CODES[_mem.rxcode]
+            else:
+                rx_tone = 0
+                rx_tmode = ""
+
+        tx_pol = txtype == 0x03 and "R" or "N"
+        rx_pol = rxtype == 0x03 and "R" or "N"
+
+        chirp_common.split_tone_decode(mem, (tx_tmode, tx_tone, tx_pol),
+                                       (rx_tmode, rx_tone, rx_pol))
+
+    # Extract a high-level memory object from the low-level memory map
+    # This is called to populate a memory in the UI
+    def get_memory(self, number2):
+        number = number2-1  # in the radio memories start with 0
+
+        mem = chirp_common.Memory()
+
+        # cutting and pasting configs from different radios
+        # might try to set channel 0
+        if number2 == 0:
```

Comment:
It's there because it has happened to some user. Unfortunately i can't reproduce it, and don't know how it could happen (this would be equivalent to the code trying to touch memory number -1).

Removed it.